tests/jballerina-unit-test/src/test/resources/test-src/expressions/varref/record-variable-reference.bal

```diff
@@ -369,6 +369,74 @@ function testMappingBindingWithSingleNameFieldBinding() {
     assertEquality("bar4", b);
 }
 
+function testMappingBindingPatternAgainstOpenRecordInTupleDestructuring() {
+    [Some, Some] r1 = [{var1: "A", var2: "B"}, {var1: "C", var2: "D"}];
+    string xVar1;
+    string xVar2;
+    [{var1: xVar1}, {var1: xVar2}] = r1;
```

Comment:
Not sure if the variable names are ideal. i.e., binding var1 to something that includes Var2 when there's a var2 field.